Test/functional/Test_State_Machine.cpp

```diff
@@ -332,13 +342,144 @@ TEST_F(StateTest, Pressure_Vessel_Depressure_Test) {
+    EXPECT_NE(getRmsCommandedTorque(), 0);
+    assertStateIs(PROPULSION_NAME);
+}
+/**
+ * Test the nominal transition from propulsion to braking
+ * Start state: propulsion        Expected state: braking
+ */
+TEST_F(StateTest, Propulsion_to_Braking_Test)
+{
+    FREEZE_SM;
+    genericInit();
+    fprintf(stderr, "[LOG] Going into Braking\n");
+    GO_TO_STATE(PROPULSION_NAME);
+    GO_TO_STATE(BRAKING_NAME);
+    UNFREEZE_SM;
+    WAIT(.5);
+
+    debuggingHelper();
+    ASSERT_EQ(getFlagsBrakePrimAct(), true); // check if brakes are primary actuated
+    assertStateIs(BRAKING_NAME);
+}
+/**
+ * Test the nominal transition from braking to stopped
+ * Start state: braking        Expected state: stopped
+ */
+TEST_F(StateTest, Braking_To_Stopped_Test)
+{
+    FREEZE_SM;
+    genericInit();
+    fprintf(stderr, "[LOG] Going into Stopped\n");
+    GO_TO_STATE(BRAKING_NAME);
+    GO_TO_STATE(STOPPED_NAME);
+    UNFREEZE_SM;
+    WAIT(.5);
+
+    debuggingHelper();
+    ASSERT_EQ(getFlagsBrakePrimRetr(), true);
+
+    assertStateIs(STOPPED_NAME);
+}
+/**
+ * Test the nominal transition from idle to pumpdown
+ * Start state: stopped        Expected state: service precharge
+ */
+TEST_F(StateTest, Stopped_to_Service_Precharge_Test)
+{
+    FREEZE_SM;
+    genericInit();
+    fprintf(stderr, "[LOG] Going into Stopped\n");
+    GO_TO_STATE(STOPPED_NAME);
+    GO_TO_STATE(SERV_PRECHARGE_NAME);
+    UNFREEZE_SM;
+    WAIT(.5);
+
+    debuggingHelper();
+    ASSERT_EQ(getFlagsShouldBrake(), false);
+    assertStateIs(SERV_PRECHARGE_NAME);
+}
+/**
+ * Test the nominal transition from idle to pumpdown
+ * Start state: service precharge         Expected state: service propulsion
+ */
+TEST_F(StateTest, Service_Precharge_To_Service_Propulsion_Test)
+{
+    FREEZE_SM;
+    genericInit();
+    fprintf(stderr, "[LOG] Going into Stopped\n");
+    GO_TO_STATE(SERV_PRECHARGE_NAME);
+    GO_TO_STATE(CRAWL_NAME);
+    UNFREEZE_SM;
+    WAIT(.5);
+
+    debuggingHelper();
+    ASSERT_EQ(getFlagsShouldBrake(), false);
+    EXPECT_NE(getRmsCommandedTorque(), 0);
+    assertStateIs(CRAWL_NAME);
+}
+/**
+ * Test the nominal transition from idle to pumpdown
+ * Start state: service propulsion        Expected state: post run
+ */
+TEST_F(StateTest, Service_Propulsion_to_Post_Run_Test)
+{
+    FREEZE_SM;
+    genericInit();
+    fprintf(stderr, "[LOG] Going into Stopped\n");
+    GO_TO_STATE(CRAWL_NAME);
+    GO_TO_STATE(POST_RUN_NAME);
+    UNFREEZE_SM;
+    WAIT(.5);
+
+    debuggingHelper();
+    assertStateIs(POST_RUN_NAME);
+}
+/**
+ * Test the nominal transition from idle to pumpdown
+ * Start state: post run        Expected state: safe to approach
+ */
+TEST_F(StateTest, Post_Run_To_Safe_To_Approach_Test)
+{
+    FREEZE_SM;
+    genericInit();
+    fprintf(stderr, "[LOG] Going into Stopped\n");
+    GO_TO_STATE(POST_RUN_NAME);
+    GO_TO_STATE(SAFE_TO_APPROACH_NAME);
+    UNFREEZE_SM;
+    WAIT(.5);
+    debuggingHelper();
```

Comment:
I like the debugging helper - maybe put it in a debug conditional compilation flag so we can easily add and remove them
```c++
#define DEBUG 1
...
#if DEBUG
debuggingHelper();
#endif
```